### PR TITLE
Add `--json` output and `--content` body rendering to `commands ls` and `skills ls`

### LINKED ISSUE #81: Add `--json`, `--full` flag support for read-only commands
## TODO

Add `--json` flag for commands like `dotagents skills ls` and `dotagents commands ls`
Add `--full` flag for commands like `dotagents skills ls` and `dotagents commands ls`

without --full flag, it only shows the command name, and its metadata (the one in front matter if exists)
with --full, it even gives out full body content

### PR DESCRIPTION
**Pull request by soorya-u**
## Summary

- Added `--json` flag to `commands ls` and `skills ls` — outputs command/skill frontmatter as a JSON array to stdout
- Added `--full` body content rendering for both text and JSON output modes
- When `--json --full` is combined, includes a `content` key with the raw markdown body string
- Extended `ListItem` struct with full frontmatter and optional body for downstream use
- All logs and warnings continue to go to stderr; JSON output is stdout-only

## Testing

- `mise check` — passed (fmt + clippy)
- `mise tests` — passed (unit + integration + e2e)
- Unit tests cover: body content inclusion, full frontmatter fields, empty dir edge case, JSON serialization with/without `--full`
- E2E tests cover: `--json` valid JSON output, `--full` body in text output, combined `--json --full`, default output (no body), empty workspace `[]`, and pipeable JSON

## Summary by CodeRabbit

* **New Features**
  * Added --json output for commands ls and skills ls (JSON array on stdout; logs/warnings to stderr)
  * Replaced --full with --content to include full markdown bodies in text output (TTY: note-style blocks; non-TTY: indented body lines)
  * Combined --json + --content includes a content field in each JSON object
  * Exact-name filters added: --command / --skill; frontmatter now exposes additional fields (category/tags/license/compatibility)

* **Documentation**
  * Updated CLI output-format and subcommand specs and design to reflect flag semantics and rendering rules

* **Tests**
  * Added unit and e2e tests covering JSON validity, empty-workspace [], flag combinations, content inclusion/omission, filtering, and pipeability (jq)

[Review Change Stack](https://app.coderabbit.ai/change-stack/soorya-u/dotagents/pull/84)

### CODE REVIEW

**Comment by soorya-u** [comment]
## Bugs & Enhancements

- [x] Replace `--full` with `--content`
- [x] There is one feature missing, We need something like `--skill` for `skill` command and `--command` for `command` command so that we can filter by ID

**Comment by soorya-u** [comment]
## Review comments resolved

All 5 CodeRabbit threads have been addressed:

**Already resolved (spec renames — committed earlier):**
- `openspec/specs/cli-output-formats/spec.md` — replaced all `--full` references with `--content`
- `openspec/specs/commands-subcommand/spec.md` — same rename throughout
- `openspec/specs/skills-subcommand-extended/spec.md` — same rename throughout

**Fixed now:**
- `src/cli/options.rs` — updated the `--content` clap doc comment from _"Show full descriptions (word-wrapped) instead of truncating"_ to _"Show full word-wrapped descriptions and include the markdown body content of each item"_, accurately reflecting that it shows body content, not just unwrapped descriptions
- `src/cli/ui/ls.rs` (`to_json_array`) — added `!body.trim().is_empty()` guard so the `content` key is only inserted into JSON output when the body has actual non-whitespace content, matching the contract that `content` is absent when body is empty

File: src/cli/ui/ls.rs [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+use console::style;
 use crossterm::terminal;
+use serde_json::Value;
 
 use crate::prelude::*;
+use crate::utils::tty::is_tty;
 
-/// A name+description pair for display.
+/// A name+description pair for display, optionally with full frontmatter + body.
 pub(crate) struct ListItem {
```
Comment (soorya-t41) [inline src/cli/ui/ls.rs]: `ListItem` should be moved to `src/schemas` and its relative implementations as well

File: src/cli/ui/ls.rs [hunk 2]
```diff
@@ -52,69 +57,102 @@ pub(crate) fn wrap_at_width(text: &str, width: usize, indent: &str) -> String {
     lines.join("\n")
 }
 
+/// Build a JSON array of frontmatter values, optionally including body content.
+pub(crate) fn to_json_array(items: &[ListItem], content: bool) -> Vec<Value> {
```
Comment (soorya-t41) [inline src/cli/ui/ls.rs]: This is the implementation that should be moved to `schemas` as well

`ListItem::to_json_array` or something close to that

File: src/cli/ui/ls.rs [hunk 3]
```diff
@@ -52,69 +57,102 @@ pub(crate) fn wrap_at_width(text: &str, width: usize, indent: &str) -> String {
...
+                && let Some(map) = obj.as_object_mut()
+            {
+                map.insert("content".to_string(), Value::String(body.clone()));
+            }
+            obj
+        })
+        .collect()
+}
+
+/// Apply cyan+bold styling to a name when colors are enabled (console handles detection).
```
Comment (soorya-t41) [inline src/cli/ui/ls.rs]: this function and other functions like terminal_cols, truncate_to_width, wrap_at_width are all common methods that could be reused. 

so move them to `src/cli/ui/common.rs`

File: src/cli/commands.rs [hunk 4]
```diff
@@ -185,17 +185,12 @@ fn rm_command(opts: RmCommandOptions) -> Result<bool> {
...
 
-/// Load all commands from `.dotagents/commands/*.md`, returning name+description pairs.
-fn load_commands() -> Result<Vec<ListItem>> {
-    let commands_dir = match get_commands_dir() {
-        Ok(d) => d,
-        Err(_) => return Ok(vec![]), // commands dir absent → empty
-    };
-
+/// Load all commands from a given directory, returning full frontmatter + body.
+fn load_commands_from(dir: &std::path::Path) -> Result<Vec<ListItem>> {
```
Comment (soorya-t41) [inline src/cli/commands.rs]: `load_commands_from` here and `load_skills_from` are almost similar. cant we create a common function for that. We can use from_application method in `src/core/features` instead of creating a new function

also when parsing the frontmatter, why are we setting it as Value while we have already defined types in `src/core/features`

File: src/cli/skills.rs [hunk 5]
```diff
@@ -253,17 +253,12 @@ fn rm_skill(opts: RmSkillOptions) -> Result<bool> {
...
 
-/// Load all skills from `.dotagents/skills/*/SKILL.md`, returning name+description pairs.
-fn load_skills() -> Result<Vec<ListItem>> {
-    let skills_dir = match get_skills_dir() {
-        Ok(d) => d,
-        Err(_) => return Ok(vec![]), // skills dir absent → empty
-    };
-
+/// Load all skills from a given directory, returning full frontmatter + body.
+fn load_skills_from(dir: &std::path::Path) -> Result<Vec<ListItem>> {
```
Comment (soorya-t41) [inline src/cli/skills.rs]: `load_skills_from` here and `load_skills_from` are almost similar. cant we create a common function for that. We can use from_application method in `src/core/features` instead of creating a new function

also when parsing the frontmatter, why are we setting it as Value while we have already defined types in `src/core/features`

File: src/cli/commands.rs [hunk 6]
```diff
@@ -254,27 +275,113 @@ mod tests {
         fs::write(dir.join(format!("{}.md", name)), content).unwrap();
     }
 
+    fn make_command_full(
```
Comment (soorya-t41) [inline src/cli/commands.rs:244]: cant we move this function to `src/core/features/command.rs` under Command::new or something

File: src/cli/skills.rs [hunk 7]
```diff
@@ -326,18 +347,88 @@ mod tests {
         fs::write(skill_dir.join("SKILL.md"), content).unwrap();
     }
 
+    fn make_skill_full(
```
Comment (soorya-t41) [inline src/cli/skills.rs:305]: cant we move this function to `src/core/features/skill.rs` under Skill::new or something

**Review by soorya-t41 — CHANGES_REQUESTED** [review]

**Comment by soorya-u** [comment]
Resolved all 8 open review threads:

- **`styled_name` / helpers → `common.rs`** (`src/cli/ui/ls.rs` line 78) — `styled_name` and `styled_note_name` moved to `src/cli/ui/common.rs` alongside `terminal_cols`, `truncate_to_width`, `wrap_at_width`.

- **Conflicting color spec** (`openspec/specs/cli-output-formats/spec.md` line 27) — Spec now explicitly scopes colors by mode: **green+bold** for `cliclack::note` headers (TTY + `--content` + non-empty body), **cyan+bold** for standard `info!` rows; non-TTY has no ANSI styling.

- **`load_commands` swallows all errors** (`src/cli/commands.rs` line 192) — Now only returns `Ok(vec![])` when the commands directory is absent (feature not enabled); all other failures (`from_application` parse/I/O errors) propagate with context.

- **`load_skills` swallows all errors** (`src/cli/skills.rs` line 250) — Same fix as above for skills.

- **Scaffold sanitization — commands** (`src/core/features/command.rs` line 95) — `CommandFeature::scaffold` now trims `category` and trims + filters empty entries from `tags` before setting `None`/`Some`.

- **Scaffold normalization — skills** (`src/core/features/skill.rs` line 88) — `SkillFeature::scaffold` now trims `license` and `compatibility` before the emptiness check, so whitespace-only inputs become `None`.

- **`wrap_at_width` byte vs char count** (`src/cli/ui/common.rs` line 49) — Changed `.len()` comparisons to `.chars().count()` so multi-byte Unicode wraps at the correct visual character boundary.

- **Missing test comments** (`src/cli/ui/common.rs` line 88) — Added single-line descriptive comments to all five tests in `common.rs`.
